designer/client/inline-conditions.js

```diff
@@ -0,0 +1,200 @@
+    }
+  }
+
+  onClickAddItem = e => {
+    this.setState({
+      condition: {}
+    })
+  }
+
+  onClickAnd = e => {
+    this.setState({
+      condition: {
+        coordinator: 'and'
+      }
+    })
+  }
+
+  onClickOr = e => {
+    this.setState({
+      condition: {
+        coordinator: 'or'
+      }
+    })
+  }
+
+  onClickSave = e => {
+    const { conditions, condition } = this.state
+
+    this.setState({
+      conditions: conditions.add(new Condition(Field.from(condition.field), condition.operator, Value.from(condition.value), condition.coordinator)),
+      condition: undefined
+    })
+  }
+
+  conditionFieldId (id, fieldName) {
+    return `conditions.${id}.${fieldName}`
+  }
+
+  parseConditionId (fieldId) {
+    return fieldId.split('.')[1]
+  }
+
+  onChangeField = e => {
+    const input = e.target
+    const fieldName = input.value
+
+    const { condition } = this.state
+
+    const currentField = condition.field ? condition.field.name : undefined
+    const currentOperator = condition.operator
+
+    this._updateCondition(condition, c => {
+      if (fieldName) {
+        if (currentField && this.fields[currentField].values !== this.fields[fieldName].values) {
+          delete c.value
+        }
+        if (currentOperator && !conditionalOperators.getConditionals(fieldName)[currentOperator]) {
+          delete c.operator
+        }
+        c.field = new Field(fieldName, this.fields[fieldName].label)
+      } else {
+        delete c.field
+        delete c.operator
+        delete c.value
+      }
+    })
+  }
+
+  _updateCondition (condition, updates) {
+    const copy = clone(condition)
+    updates(copy)
+    this.setState({
+      condition: copy
+    })
+  }
+
+  onChangeOperator = e => {
+    const input = e.target
+    const { condition } = this.state
+
+    this._updateCondition(condition, c => { c.operator = input.value })
+  }
+
+  onChangeValue = e => {
+    const input = e.target
+    const { condition } = this.state
+
+    this._updateCondition(condition, c => { c.value = new Value(input.value) })
+  }
+
+  render () {
+    const { conditions, condition } = this.state
+
+    const fieldDef = condition && condition.field ? this.fields[condition.field.name] : undefined
+
+    const hasConditions = conditions.hasConditions()
+
+    return (
+      <div>
+        { hasConditions &&
+          <div className='govuk-form-group'>
+            <label className='govuk-label govuk-label--s' htmlFor='condition-string'>When</label>
+            <div key='condition-string' >
+              {conditions.toPresentationString()}
+            </div>
+          </div>
+        }
+        { condition &&
+        <div className='govuk-form-group'>
+          <label className='govuk-label govuk-label--s' htmlFor='field'>{hasConditions ? condition.coordinator : 'When'}</label>
+          <select className='govuk-select' id='cond-field' name='cond-field' value={condition && condition.field ? condition.field.name : undefined}
```

Comment:
`condition && condition.field ? condition.field.name` this is also another reason why I want ES2020. maybe for another ticket though! (Would just be `condition?.field?.name` and no need for ternary.)

Reply:
Yep, that's a much nicer syntax, I'm not entirely sure how to go about changing to ES2020 though,  looks like maybe updating the version of babel and we get it automagically?

Reply:
yup. I think rollup will need an update to ^2.14.0 if I remember correctly as well. Happy for this not to block your merge unless you want to implement.